File: my_package/move_turtle.py
import rclpy
from rclpy.node import Node
from geometry_msgs.msg import Twist
from rclpy.qos import QoSProfile
from turtlesim.msg import Pose

class M_turtle(Node):
  def __init__(self):
    super().__init__('move_turtle')
    self.qos = QoSProfile(depth = 10)
    self.pub = self.create_publisher(Twist, 'turtle1/cmd_vel', self.qos)
    self.sub = self.create_subscription(Pose, '/turtle1/pose', self.subpose, self.qos)
    self.create_timer(0.1, self.pubmessage)
    self.vel = 0.0
    self.x = 0.0
    self.y = 0.0
    self.theta = 0.0

  def pubmessage(self):
    msg = Twist()
    msg.linear.x = self.vel
    msg.linear.y = 0.0
    msg.linear.z = 0.0
    msg.angular.x = 0.0
    msg.angular.y = 0.0
    msg.angular.z = 2.5
    self.pub.publish(msg)
    # self.get_logger().info(f'Seding message: [{msg}]')
    self.vel += 0.04
    if self.vel > 3.0:
      self.vel = 0.0

  def subpose(self, msg):
    self.x = msg.x
    self.y = msg.y
    self.theta = msg.theta

def main():
  rclpy.init()
  node = M_turtle()
  try:
    rclpy.spin(node)
  except:
    node.destroy_node()

if __name__ == '__main__':
  main()

File: my_package_cpp/src/move_turtle.cpp
#include <chrono>
#include <functional>
#include <memory>
#include <string>

#include "rclcpp/rclcpp.hpp"
#include "geometry_msgs/msg/twist.hpp"

using namespace std::chrono_literals;

class HelloworldPublisher : public rclcpp::Node
{
private:
    rclcpp::TimerBase::SharedPtr timer_;
    rclcpp::Publisher<geometry_msgs::msg::Twist>::SharedPtr turtle_publisher_;
    size_t count_;
    void publish_helloworld_msg()
    {
        auto msg = geometry_msgs::msg::Twist();
        msg.linear.x = 1.0;
        msg.angular.z = 0.4;
        turtle_publisher_->publish(msg);
    }

public:
    HelloworldPublisher()
    : Node("Helloworld_publisher"), count_(0)
    {
        auto qos_profile = rclcpp::QoS(rclcpp::KeepLast(10));
        turtle_publisher_ = this->create_publisher<geometry_msgs::msg::Twist>(
            "turtle1/cmd_vel", qos_profile);
        timer_ = this->create_wall_timer(
            0.1s, std::bind(&HelloworldPublisher::publish_helloworld_msg, this));
    }

};

int main(int argc, char * argv[])
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<HelloworldPublisher>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}